module/module.go

```diff
@@ -268,19 +253,21 @@ func (m *Module) GetParentResource(ctx context.Context, name resource.Name) (res
 }
 
 func (m *Module) connectParent(ctx context.Context) error {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	if m.parent == nil {
-		if err := CheckSocketOwner(m.parentAddr); err != nil {
-			return err
-		}
-		// TODO(PRODUCT-343): add session support to modules
-		rc, err := client.New(ctx, "unix://"+m.parentAddr, m.logger, client.WithDisableSessions())
-		if err != nil {
-			return err
-		}
-		m.parent = rc
+	if err := CheckSocketOwner(m.parentAddr); err != nil {
+		return err
+	}
+
+	// NOTE(benjirewis): moduleLoggers may be creating the client connection
```

Comment:
[minor] no need to attribute since we have git blame - just a little less noise :)

```suggestion
	// moduleLoggers may be creating the client connection
```